proxy/proxy.go

```diff
@@ -1526,3 +1529,35 @@ func ensureUTF8(s string) string {
 	}
 	return fmt.Sprintf("invalid utf-8: %q", s)
 }
+
+// Opt-In for filters to get called Response(ctx) in case of errors
+type errorHandlerFilter interface {
+	HandleErrorResponse()
+}
+
+func (p *Proxy) handleFiltersOnError(ctx *context, filters []*routing.RouteFilter) {
```

Comment:
Lets name it `applyFiltersOnError` similar to existing `applyFiltersToRequest` and `applyFiltersToResponse`

Reply:
done